### PR TITLE
Add DISA STIG ids to `when` conditions in ansible roles

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Add DISA STIG ids to `when` conditions in ansible roles.

#### Rationale:

- Tasks can be controlled by the STIG id.

Fixes: https://github.com/ComplianceAsCode/content/issues/7092 in commit ff5d600328f9ec4e5de248e003d349c24903bad4

### CODE REVIEW

**Comment by jan-cerny** [comment]
Hi, what is needed here to move it forward?

**Comment by ggbecker** [comment]
> Hi, what is needed here to move it forward?

We need to decide if that's a good long term solution approach. And, do we want to include other references as well or only STIG ones?

File: utils/ansible_playbook_to_role.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def added_variables(self):
             elif isinstance(task["when"], str):
                 task["when"] = [task["when"]]
 
-            variables_to_add = {tag for tag in task["tags"] if self._tag_is_valid_variable(tag)}
+            variables_to_add = {self._sanitize_tag(tag) for tag in task["tags"] if self._tag_is_valid_variable(tag)}
```
Comment (Mab879) [inline utils/ansible_playbook_to_role.py:214]: PEP8 is complaining about this line. We may want to rewrite this for clarity.

**Review by Mab879 — CHANGES_REQUESTED** [review]

File: utils/ansible_playbook_to_role.py (hunk 1 above)
Comment (ggbecker) [reply utils/ansible_playbook_to_role.py:214]: I know, it's about the line length. But the problem was there before and many other lines go beyong the 99 maximum line length. I refactored it to less than 120 chars, but 99 seems too little.

Comment (ggbecker) [reply utils/ansible_playbook_to_role.py]: Ok, I have broken some lines and it seems the issue is gone.

**Comment by ggbecker** [comment]
I went further and fixed: #7092 which was a very simple fix.

**Review by Mab879 — APPROVED** [review]
Checking and I was able to create the role, and the role did start executing.

**Pull request merged**